### PR TITLE
Changed saving of colors as strings to an rgb tuple

### LINKED ISSUE #76: Save colormaps as 'cmap' in field arrays instead of 'color'
The load method should also be updated for this change accordingly as we have to check for the existence of a `cmap` in the field array.
@dstansby Should we warn the user if a color value as well as a cmap is passed to add_mesh? In pyvista, the color argument takes preference.

### PR DESCRIPTION
**Pull request by jeffreypaul15**
Fixes #76  
Previously, field_arrays contained the name of the color as the 0th index of the `color` field array.
I've changed this to always have a tuple of type (r,g,b) as pyvista accepts this format for colors as well.

The default value of color is now always white - previously being `nan`. This avoids unnecessary checks when loading the colors.

### CODE REVIEW

**Comment by jeffreypaul15** [comment]
@dstansby Previously when a line was plotted, the arc_length legend was shown to denote the length in terms of solar radius, since we're using white as the default color, this is replaced with a white line. Do we need the previous functionality?
image
image

**Comment by dstansby** [comment]
I think having the line as white is actually better 😄

File: changelog/73.trivial.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Changes the way colors are saved as well as default colors of meshes.
```
Comment (nabobalis) [inline changelog/73.trivial.rst:1]:
```suggestion
Changed the manner that colors or colormaps are saved.
Changed default of meshes to be white.
```

File: sunkit_pyvista/plotter.py [hunk 2]
```diff
@@ -420,14 +428,10 @@ def _loop_through_meshes(self, mesh_block):
...
             else:
-                color = dict(block.field_arrays).pop('color', [None])[0]
-
-                if not isinstance(color, str):
-                    color = None
-                if color in _cmap_registry:
-                    self.plotter.add_mesh(block, cmap=color)
-                else:
-                    self.plotter.add_mesh(block, color=color)
+                print(dict(block.field_arrays))
```
Comment (nabobalis) [inline sunkit_pyvista/plotter.py:431]:
```suggestion
```

File: sunkit_pyvista/plotter.py [hunk 3]
```diff
@@ -218,9 +219,13 @@ def plot_map(self, m, clip_interval: u.percent = None, **kwargs):
         else:
             clim = [0, 1]
         cmap = self._get_cmap(kwargs, m)
-        self.plotter.add_mesh(map_mesh, cmap=cmap, clim=clim, **kwargs)
+        color = kwargs.pop('color', None)
```
Comment (dstansby) [inline sunkit_pyvista/plotter.py:222]: Sorry, I was wrong in my earlier comment - we shouldn't allow color as a keyword argument to `plot_map`, as it makes no sense to plot a map in a single color. Instead just leave this line in, add a warning below in the `if` block, and don't add color to the field array.

File: sunkit_pyvista/plotter.py [hunk 4]
```diff
@@ -304,9 +310,10 @@ def plot_solar_axis(self, length=2.5, arrow_kwargs={}, **kwargs):
                               direction=(0, 0, length),
                               scale='auto',
                               **defaults)
-        color = kwargs.get('color', np.nan)
-        arrow_mesh.add_field_array([color], 'color')
-        self.plotter.add_mesh(arrow_mesh, **kwargs)
+        color = kwargs.pop('color', 'white')
```
Comment (dstansby) [inline sunkit_pyvista/plotter.py:313]: Could you create a new method called `_extract_color` that:

- Takes `kwargs` as an input
- Returns a color as an RGB tuple?

This was we can reduce code duplication in several of the new lines you've added.

Comment (jeffreypaul15) [reply sunkit_pyvista/plotter.py:313]: Yes, this makes more sense I shall do that

File: sunkit_pyvista/plotter.py (hunk 3 above)
Comment (jeffreypaul15) [reply sunkit_pyvista/plotter.py:222]: Alright, I'll make this change.

File: sunkit_pyvista/plotter.py (hunk 4 above)
Comment (jeffreypaul15) [reply sunkit_pyvista/plotter.py:313]: Yes I'll add thsi in.

**Review by dstansby — APPROVED** [review]

**Pull request merged**